build/codegen/codegen.sh

```diff
@@ -14,29 +14,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+set -o pipefail
+set -o nounset
+set -o errexit
+
+
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+base_dir="$(cd "${script_dir}/../.."  && pwd)"
+
+boilerplate="${base_dir}"/build/codegen/header.txt
+
+
+# echo "DEBUG: script_dir: ${script_dir}"
+# echo "DEBUG: CODE_GENERATOR: ${CODE_GENERATOR}"
+# echo "DEBUG: boilerplate ${boilerplate}"
+#  echo "DEBUG: done"
+# exit 0
+
+
+
+
+
+source "${CODE_GENERATOR}/kube_codegen.sh"
+
+
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # CODE GENERATION
 # we run deepcopy and client,lister,informer generations separately so we can use the flag "--plural-exceptions"
 # which is only known by client,lister,informer binary and not the deepcopy binary
 
 # run code deepcopy generation
-bash ${CODE_GENERATOR}/generate-groups.sh \
-    deepcopy \
-    github.com/rook/rook/pkg/client \
-    github.com/rook/rook/pkg/apis \
-    "${GROUP_VERSIONS}" \
-    --output-base "$(dirname "${BASH_SOURCE[0]}")/../../../../.." \
-    --go-header-file "${scriptdir}/boilerplate.go.txt"
+ kube::codegen::gen_helpers \
+    --boilerplate "${boilerplate}" \
+    "github.com/rook/rook/pkg/apis" \
```

Comment:
Directory location, not package location

```suggestion
    "${base_dir}/pkg/apis" \
```